### PR TITLE
[ARTEMIS-1527] - [Artemis Testsuite] ActiveMQMessageHandlerTest#testS…

### PR DESCRIPTION
**Pull request by JiriOndrusek**
…erverShutdownAndReconnect fails

Issue: https://issues.apache.org/jira/browse/ARTEMIS-1527

### CODE REVIEW

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/RemotingConnectionImpl.java [hunk 1]
```diff
@@ -205,14 +211,59 @@ public void fail(final ActiveMQException me, String scaleDownTargetNodeID) {
...
+                  running.add(l);
+               }
+            } catch (InterruptedException e) {
+               ActiveMQClientLogger.LOGGER.warn(e.getMessage(), e);
+            }
+         });
+
+         if (!running.isEmpty()) {
+            //TODO(jondruse) is there constant or propertyF usable for this kind of wait interval?
+            scheduledExecutorService.schedule(new ScheduledInternalClose(running, me, latch), 500, TimeUnit.MILLISECONDS);
```
Comment (JiriOndrusek) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/RemotingConnectionImpl.java]: I have too use "constant" for repetitive checking, whether action is already finished. May I use some generic timeout used in project (or define new attribute, ...)

**Comment by JiriOndrusek** [comment]
In 2 places I have to use "constant" for repetitive checking, whether action is already finished. May I use some generic timeout used in project (or define new attribute, ...)
(ClientSessionFactoryImpl line 1123 and RemotingConnectionImpl line 254)

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionFactoryImpl.java [hunk 2]
```diff
@@ -990,17 +1111,26 @@ public CloseRunnable(RemotingConnection conn, String scaleDownTargetNodeID) {
...
-            CLOSE_RUNNABLES.remove(this);
+         CLOSE_RUNNABLES.add(this);
+         CountDownLatch latch;
+         if (scaleDownTargetNodeID == null) {
+            latch = conn.fail(ActiveMQClientMessageBundle.BUNDLE.disconnected());
+         } else {
+            latch = conn.fail(ActiveMQClientMessageBundle.BUNDLE.disconnected(), scaleDownTargetNodeID);
          }
 
+         //TODO(jondruse) is there constant or property usable for this kind of wait interval?
```
Comment (JiriOndrusek) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionFactoryImpl.java]: I have too use "constant" for repetitive checking, whether action is already finished. May I use some generic timeout used in project (or define new attribute, ...)

**Comment by clebertsuconic** [comment]
-1000 on this PR...

it's a brittle change over a test failure. .you could find other way to fix the test?


it's a huge change on the maintenance 1.x branch... it's a risky change... if this is a go... it needs to be on master. (I wouldn't ever cherry-pick something this size into 1.x anyways).

**Comment by jbertram** [comment]
@JiriOndrusek, I don't think this is going to get merged. Can you go ahead and close this PR?

**Comment by JiriOndrusek** [comment]
I agree, closing as requested in previous comment.

**Pull request closed without merging**